docs/changes/table/18641.feature.rst

```diff
@@ -0,0 +1,9 @@
+The ``read()`` methods of ``QTable`` and ``Table`` have been made more
+flexible when reading tables with optional columns that may or may not
+appear in the data.
+
+These methods accept ``units`` and ``description`` arguments as a
+``dict`` that specifies the units and descriptions for column names in
+the table. Previously, if the input table did not contain a column that
+was specified in the ``units`` or ``description`` dict, a
+``ValueError`` was raised. Now, such columns are simply ignored.
```

Comment:
I read this again today and realized that it is more general since it applies to table creation with the class initializer as well as the `read()` method.
```suggestion
Setting the ``units`` or ``descriptions`` of ``QTable`` and ``Table``
has been made more flexible for tables with optional columns that may
or may not appear in the data. This applies to directly creating a table 
as well as reading formatted data with the ``read()`` method.

In both cases you can supply ``units`` and ``description`` arguments as a
``dict`` that specifies the units and descriptions for column names in
the table. Previously, if the input table did not contain a column that
was specified in the ``units`` or ``description`` dict, a ``ValueError`` 
was raised. Now, such columns are simply ignored.
```